src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -57,6 +62,48 @@
   @Requirement
   private IModelAccessFacade modelAccess;
 
+  @Requirement
+  private List<IPageTypeCategoryRole> pageTypeCategoryList;
+
+  private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCategoryMap = new ConcurrentHashMap<>();
+
+  private ConcurrentMap<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+    if (typeNameToCategoryMap.isEmpty()) {
+      synchronized (typeNameToCategoryMap) {
+        if (typeNameToCategoryMap.isEmpty()) {
+          for (IPageTypeCategoryRole typeCategory : pageTypeCategoryList) {
+            IPageTypeCategoryRole beforeRegCat = typeNameToCategoryMap.putIfAbsent(
+                typeCategory.getTypeName(), typeCategory);
```

Comment:
I don't see the need for a ConcurrentMap since there is no simultaneous read/write to the map. Assigning a volatile variable inside synchronized should be sufficient.
I'd still consider immutability for clarity reasons: the map is not expected to change outside the synchronized block, therefore fast failure upon modification is preferable imho.